src/common/SurgeLogger.cpp

```diff
@@ -0,0 +1,57 @@
+#include "SurgeLogger.h"
+#include "spdlog/sinks/basic_file_sink.h"
+#include "spdlog/sinks/stdout_sinks.h"
+
+#include "spdlog/logger.h"
+#include "spdlog/async.h"
+#include <memory>
+
+
+typedef std::shared_ptr< spdlog::logger > log_t;
+
+log_t internal_surge_log_init()
+{
+  std::vector<spdlog::sink_ptr> sinks;
+  
```

Comment:
Not really a review comment but should we just move to "more standard" 4 character tabs for new code :D

Reply:
To get more coheren codebase I would use CamelCase for all type (e.g. `Log`) and camelCase for or variables and function (e.g. `internalSurgeLogInit()`). Now this contributes to the porridge of different styles and conventions. Lets get moro rigid.

Reply:
Not really a review comment but maybe we could have

```
namespace Log {
    Log init()
    {
    }
}
```

We could possibly have few namespaces e.g. Log, GUI, Synth (non-GUI engine). I don't really get why you wan't to prefix with "Surge" and for me anything that does not have static keyword is not internal. It is exported symbol.

Reply:
Being programmed with only C for past 8-9 C++ is a mystery box but didn't recent standard versions have some kind lambda type of thing, or am I wrong? Then you could initialize the log by using such?

Reply:
Yeah c++ 11 lambdas are cool - I used them in zoom - but they aren’t the right construxtnfor this sort of initializarion. 

Like the namespace idea. Will do that. And will clean up all the casing and stuff. A lot of this got copied from the spdlog manual which has yet another coding style.

The mush mash of styles and tools is driving me crazy too. We should adopt a standard!

Reply:
Oh the only one of your comments I'm not integrating is the type rename. stl has a pretty clear convention that types are named like `thething_t` so I think `typedef thisWhatzIt<foobaz> mything_t` is very idiomatic. Just an FYI for when you read the next version!